server/routes/utils/dataReportHelpers.ts

```diff
@@ -0,0 +1,212 @@
+import { async } from 'rxjs/internal/scheduler/async';
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { DATA_REPORT_CONFIG } from './constants';
+
+const esb = require('elastic-builder');
+const moment = require('moment');
+const converter = require('json-2-csv');
+
+export var metaData = {
+	saved_search_id: <string>null,
+	report_format: <string>null,
+	start: <string>null,
+	end: <string>null,
+	fields: <string>null,
+	type: <string>null,
+	timeFieldName: <string>null,
+	sorting: <string>null,
+	fields_exist: <boolean>false,
+	selectedFields: <any>[],
+	paternName: <string>null,
+	filters: <any>[],
+	dateFields: <any>[],
+};
+
+// Get the selected columns by the user.
+export const getSelectedFields = async (columns) => {
+	for (let column of columns) {
+		if (column !== '_source') {
+			metaData.fields_exist = true;
+			metaData.selectedFields.push(column);
+		} else {
+			metaData.selectedFields.push('_source');
+		}
+	}
+};
+
+//Build the ES query from the meta data
+// is_count is set to 1 if we building the count query but 0 if we building the fetch data query
+export const buildQuery = (report, is_count) => {
```

Comment:
a simple boolean value will make is more readable.

Reply:
yes it's a boolean

Reply:
I mean the variable "is_count". You are using 0/1 instead of true/false. It's up to you, but I think for a statement "is_count = 1", the readability is not very good.